allennlp/data/vocabulary.py

```diff
@@ -9,26 +9,37 @@
 
 class _NamespaceDependentDefaultDict(defaultdict):
     """
-    Using ``defaultdicts`` to handle namespaces is really convenient - you don't need to worry
-    about whether you've seen a particular namespace before in any of the methods in
-    ``Vocabulary``.  But because some namespaces need padding (like "tokens") and some don't (like
-    "labels"), we want different defaults depending on the namespace.  This lets us still use a
-    ``defaultdict``, but have different default values depending on the key that we're given.
+    Sometimes certain namespaces need padding (like "tokens") and some don't (like
+    "labels"), and we want different defaults depending on the namespace.  This class lets us use a
+    ``defaultdict`` (https://docs.python.org/2/library/collections.html#collections.defaultdict),
+    but have different default values depending on the namespace of the key.
+
+    This class also handles *-namespaces.  In other words, if "*tags" is in non_padded_namespaces
+    then "passage_tags", "question_tags", etc. (anything that ends with "tags" will have the
+    non_padded default value.
```

Comment:
We should document the parameters here.  I didn't, because it's a private class, but I'll do it.